theseus/core/loss_function.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+from typing import List, Optional, Tuple, cast
+
+import torch
+
+from theseus.core.theseus_function import TheseusFunction
+
+from ..geometry import Vector
+
+
+class LossFunction(TheseusFunction, abc.ABC):
+    def __init__(self, name: Optional[str] = None):
+        super().__init__(name=name)
+
+    @abc.abstractmethod
+    def function_value(self, error: torch.Tensor) -> torch.Tensor:
+        pass
+
+    @abc.abstractmethod
+    def rescale(
+        self,
+        jacobians: List[torch.Tensor],
+        error: torch.Tensor,
+    ) -> Tuple[List[torch.Tensor], torch.Tensor]:
+        pass
+
+    def copy(
+        self, new_name: Optional[str] = None, keep_variable_names: bool = False
+    ) -> "LossFunction":
+        return cast(
+            LossFunction,
+            super().copy(new_name=new_name, keep_variable_names=keep_variable_names),
+        )
+
+
+class TrivialLoss(LossFunction):
+    def __init__(self, name: Optional[str] = None):
+        super().__init__(name)
+
+    def function_value(self, error: torch.Tensor) -> torch.Tensor:
+        return torch.sum(error**2, dim=1, keepdim=True)
+
+    def rescale(
+        self,
+        jacobians: List[torch.Tensor],
+        error: torch.Tensor,
+    ) -> Tuple[List[torch.Tensor], torch.Tensor]:
+        return jacobians, error
+
+    def _copy_impl(self, new_name: Optional[str] = None) -> "TrivialLoss":
+        return TrivialLoss(name=new_name)
+
+    # only added to avoid casting downstream
+    def copy(
+        self, new_name: Optional[str] = None, keep_variable_names: bool = False
+    ) -> "TrivialLoss":
+        return cast(
+            TrivialLoss,
+            super().copy(new_name=new_name, keep_variable_names=keep_variable_names),
+        )
+
+
+class RobustLoss(LossFunction):
```

Comment:
Since you have abstract methods in this class
`class RobustLoss(LossFunction, abc.ABC)`